packages/data-service/src/data-service.ts

```diff
@@ -1513,6 +1534,51 @@ class DataServiceImpl extends WithLogContext implements DataService {
     return await coll.dropIndex(name);
   }
 
+  @op(mongoLogId(1_001_000_237))
+  async isListSearchIndexesSupported(ns: string): Promise<boolean> {
+    try {
+      await this.getSearchIndexes(ns);
+    } catch (err) {
+      return false;
+    }
+    return true;
+  }
+
+  @op(mongoLogId(1_001_000_238))
+  async getSearchIndexes(ns: string): Promise<SearchIndex[]> {
+    const coll = this._collection(ns, 'CRUD');
+    const cursor = coll.listSearchIndexes();
+    const indexes = await cursor.toArray();
+    void cursor.close();
+    return indexes as SearchIndex[];
+  }
+
+  @op(mongoLogId(1_001_000_239))
+  createSearchIndex(
```

Comment:
I know it doesn't change anything, but should we change this to async if it's returning a Promise? Just to keep the conventions.

Reply:
I guess to much the others in the file. I'm in two minds about it because the return value is a Promise and some lint rules would even complain that this function doesn't use await. I guess I might as well change it and keep everything consistent.